balena/models/device.py

```diff
@@ -1033,7 +1033,41 @@ def set_to_release(self, uuid, commit_id):
         """
 
         device_id = self.get(uuid)['id']
-        release_id = self.release._Release__get_by_option('commit', commit_id)[0]['id'] if commit_id else None
+        release_id = self.release._Release__get_by_option('commit', commit_id, status='success')[0]['id'] if commit_id else None
```

Comment:
Keep in mind that this might return a release with the same commit hash from a different application.
Ideally we should also filter or check that the retrieved release is of the same application as the device.

Reply:
in essence, the whole pin to by commit hash, is somewhat unpredictable, i really prefer to use the set_ by_release_id, which is unique and stable

Reply:
@thgreasi added filter for application.

@amirfuhrmann release id is not something you can get from the dashboard so I think it's better to support both commit hash and release id.

Reply:
You can actually get if from the url, but anyway.